tests/fixtures/unicode-property-escape.js

```diff
@@ -1,4 +1,208 @@
+	{
+		pattern: '\\p{ASCII_Hex_Digit}\\P{ASCII_Hex_Digit}',
+		options: {},
+		expected: '\\p{ASCII_Hex_Digit}\\P{ASCII_Hex_Digit}'
+	},
+
+	// throws on unknown binary properties
+	{
+		pattern: '\\p{UnknownBinaryProperty}',
+		throws: /Unknown property: UnknownBinaryProperty/
+	},
+	{
+		pattern: '\\P{UnknownBinaryProperty}',
+		throws: /Unknown property: UnknownBinaryProperty/
+	},
+	// throws on explicitly unsupported properties
+	// https://github.com/tc39/proposal-regexp-unicode-property-escapes/issues/27
+	{
+		pattern: '\\P{Composition_Exclusion}',
+		throws: /Unknown property: Composition_Exclusion/
+	},
+	{
+		pattern: '\\p{Expands_On_NFC}',
+		throws: /Unknown property: Expands_On_NFC/
+	},
+	{
+		pattern: '\\p{Expands_On_NFD}',
+		throws: /Unknown property: Expands_On_NFD/
+	},
+	{
+		pattern: '\\p{Expands_On_NFKC}',
+		throws: /Unknown property: Expands_On_NFKC/
+	},
+	{
+		pattern: '\\p{Expands_On_NFKD}',
+		throws: /Unknown property: Expands_On_NFKD/
+	},
+	{
+		pattern: '\\p{FC_NFKC_Closure}',
+		throws: /Unknown property: FC_NFKC_Closure/
+	},
+	{
+		pattern: '\\p{Full_Composition_Exclusion}',
+		throws: /Unknown property: Full_Composition_Exclusion/
+	},
+	{
+		pattern: '\\P{Grapheme_Link}',
+		throws: /Unknown property: Grapheme_Link/
+	},
+	{
+		pattern: '\\P{Hyphen}',
+		throws: /Unknown property: Hyphen/
+	},
+	{
+		pattern: '\\P{Other_Alphabetic}',
+		throws: /Unknown property: Other_Alphabetic/
+	},
+	{
+		pattern: '\\P{Other_Default_Ignorable_Code_Point}',
+		throws: /Unknown property: Other_Default_Ignorable_Code_Point/
+	},
+	{
+		pattern: '\\P{Other_Grapheme_Extend}',
+		throws: /Unknown property: Other_Grapheme_Extend/
+	},
+	{
+		pattern: '\\P{Other_ID_Continue}',
+		throws: /Unknown property: Other_ID_Continue/
+	},
+	{
+		pattern: '\\P{Other_ID_Start}',
+		throws: /Unknown property: Other_ID_Start/
+	},
+	{
+		pattern: '\\P{Other_Lowercase}',
+		throws: /Unknown property: Other_Lowercase/
+	},
+	{
+		pattern: '\\P{Other_Math}',
+		throws: /Unknown property: Other_Math/
+	},
+	{
+		pattern: '\\P{Other_Uppercase}',
+		throws: /Unknown property: Other_Uppercase/
+	},
+	{
+		pattern: '\\P{Prepended_Concatenation_Mark}',
+		throws: /Unknown property: Prepended_Concatenation_Mark/
+	},
+	// throws on non-binary properties without a value
+	{
+		pattern: '\\p{General_Category}',
+		throws: /Failed to recognize value `undefined` for property `General_Category`\./
+	},
+	// throws on unknown property values
+	{
+		pattern: '\\p{General_Category=UnknownCategory}',
+		throws: /Unknown value `UnknownCategory` for property `General_Category`\./
+	},
+	{
+		pattern: '\\P{General_Category=UnknownCategory}',
+		throws: /Unknown value `UnknownCategory` for property `General_Category`\./
+	},
+	// throws when loose matching is attempted
+	{
+		pattern: '\\p{gc=uppercaseletter}',
+		throws: /Unknown value `uppercaseletter` for property `General_Category`\./
+	},
+	{
+		pattern: '\\p{Block=Superscripts and Subscripts}',
```

Comment:
Previously we are not properly testing the desired failure as the leading slash is not escaped:

https://github.com/mathiasbynens/regexpu-core/blob/9e1e86b74c262d09afaf3e0a943c127d6f77d1e6/tests/tests.js#L233

This is now fixed.